Octokit.Tests.Integration/Reactive/ObservableCommitStatusClientTests.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System.Reactive.Linq;
+using System.Threading.Tasks;
+using Octokit.Reactive;
+using Xunit;
+
+namespace Octokit.Tests.Integration.Reactive
+{
+    public class ObservableCommitStatusClientTests
+    {
+        public class TheGetAllMethod
+        {
+            readonly ObservableCommitStatusClient _commitStatusClient;
+            const string owner = "octokit";
+            const string name = "octokit.net";
+            const string reference = "master";
+
+            public TheGetAllMethod()
+            {
+                var github = Helper.GetAuthenticatedClient();
+                _commitStatusClient = new ObservableCommitStatusClient(github);
+            }
+
+            [IntegrationTest]
+            public async Task ReturnCommitStatus()
+            {
+                var commitStatus = await _commitStatusClient.GetAll(owner, name, reference).ToList();
+
+                Assert.NotEmpty(commitStatus);
+            }
+
+            [IntegrationTest]
+            public async Task ReturnsCorrectCountOfCommitStatusWithoutStart()
+            {
+                var options = new ApiOptions
+                {
+                    PageSize = 5,
```

Comment:
Given we're working with five commit statuses, you could change this `PageSize` to 2 so that you can get these other tests working as expected...